webpack/assets/javascripts/react_app/common/AwaitedMount.js

```diff
@@ -20,32 +17,22 @@ export const AwaitedMount = ({ component, data, flattenData }) => {
           flattenData,
         })
       );
-    } else if (allPluginsImported) {
-      const awaitedComponent = componentRegistry.markup(component, {
-        data,
-        store,
-        flattenData,
-      });
-      setMounted(true);
-      setMountedComponent(awaitedComponent);
+    } else {
+      // eslint-disable-next-line no-console
+      console.debug(
+        `Component not found: ${component}. The script for the component might not have been loaded yet.`
+      );
     }
-  }
-  const updateAllPluginsImported = e => {
-    setAllPluginsImported(true);
-  };
-  useEffect(() => {
-    document.addEventListener('loadJS', updateAllPluginsImported);
-    return () => window.removeEventListener('loadJS', updateAllPluginsImported);
-  }, []);
+  }, [component, data, flattenData]);
   useEffect(() => {
-    if (!mounted) mountComponent();
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [allPluginsImported]);
+    document.addEventListener('loadPlugin', mountComponent);
```

Comment:
I don't understand how this is listening for all plugins' JS to be loaded and not just a single random plugin?

Reply:
Its listening to each plugin load, and then checking if the component is registered. if it is it loads it.